crates/communication/src/server/runtime.rs

```diff
@@ -60,25 +84,73 @@ impl Runtime {
 
                 let inner_runtime = runtime.clone();
                 runtime.block_on(async move {
+                    let initial_parameters: Parameters =
+                        match deserialize(&parameters_directory, &body_id, &head_id).await {
+                            Ok(initial_parameters) => initial_parameters,
+                            Err(source) => {
+                                runtime_sender.send(None).expect(
+                                "successful thread creation should always wait for runtime_sender",
+                            );
+                                return Err(StartError::InitialParametersNotParsed(source));
+                            }
+                        };
+
                     let (outputs_sender, outputs_receiver) = channel(1);
+
+                    let (parameters_writer, parameters_reader) = multiple_buffer_with_slots(
+                        repeat_with(|| initial_parameters.clone())
+                            .take(amount_of_parameters_slots + 1),
+                    );
+
+                    let (parameters_sender, parameters_receiver) = channel(1);
+                    let parameters_changed = Arc::new(Notify::new());
+                    let (parameters_storage_sender, parameters_storage_receiver) = channel(1);
+
                     runtime_sender
-                        .send(Some((inner_runtime, outputs_sender.clone())))
+                        .send(Some((
+                            inner_runtime,
+                            outputs_sender.clone(),
+                            DebugWrapper(parameters_reader.clone()),
+                        )))
                         .expect("successful thread creation should always wait for runtime_sender");
```

Comment:
This `DebugWrapper` seems super hacky.
You need this because `.expect` wants to print the "error", which in the case of a send operation contains the message that was failed to be sent, right?
Since as far as I can see, the Debug implementations of the other types also don't contain any useful information, I would suggest reducing the `Result` to an `Option` via `.ok()` and expecting that instead.

Reply:
I think it is not that bad to wrap the type to be able to implement a trait differently. This is how Rust type composition works. But since it is fine to use `.ok()`, I removed the `DebugWrapper`.